server.js

```diff
@@ -13,13 +13,94 @@ mongoose.Promise = Promise
 const port = process.env.PORT || 8080
 const app = express()
 
+//schema which defines the data which is posted to the backend
+const ThoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: true,
+    minlength: [
+      5,
+      "Please type in more words",
+    ],
+    maxlength: [
+      140, 
+      "Please stop writing; that are enough words."
+    ],
+    trim: true,
+  },
+
+  createdAt: {
+    type: Date,
+    default: Date.now,
+    required: true,
+  },
+
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+})
+
+const Thought = mongoose.model('Thought', ThoughtSchema)
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(express.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Hello my dear people out there, this is my API to the happy thought project')
+})
+
+// endpoint for users to fetch the most recent 20 thoughts
+app.get("/thoughts", async (req, res) => {
+  try {
+    const allThoughts = await Thought.find()
+      .sort({ createdAt: "desc" })
+      .limit(20)
+    res.status(200).json(allThoughts)
+  } catch (error) {
+    res.status(404).json({
+      message: "Cannot find thoughts",
+      errors: error.errors,
+      success: false,
+    })
+  }
+})
+
+app.post('/thoughts', async (req, res) => {
+  const { message, name } = req.body
+
+  try {
+    const newThought = await new Thought({ message, name }).save()
+    res.status(201).json({ response: newThought, success: true})
+  } catch (error) {
+    res.status(400).json({ response: error, success: false})
+  }
```

Comment:
On line 78: I think you could put a message with the json, something like message : "could not save thought to the database"